python/tank/authentication/sso_saml2/core/utils.py

```diff
@@ -86,20 +86,44 @@ def _get_shotgun_user_id(cookies):
     user_id = None
     user_domain = None
     for cookie in cookies:
-        # Shotgun appends the unique numerical ID of the user to the cookie name:
-        # ex: shotgun_sso_session_userid_u78
-        if cookie.startswith("shotgun_sso_session_userid_u"):
+        if cookie.startswith("csrf_token_u") or cookie == "shotgun_current_user_id":
```

Comment:
It seems you are introducing a regression here. If somebody upgrades Toolkit but not Shotgun the code will break because `shotgun_sso_session_userid_u` will still be in the cookies.

Reply:
Local sites, which generally take a time before updating Shotgun, would break for example.